prism_compile.c

```diff
@@ -2790,7 +2790,7 @@ pm_add_ensure_iseq(LINK_ANCHOR *const ret, rb_iseq_t *iseq, int is_return, const
 static void
 pm_insert_local_index(pm_constant_id_t constant_id, int local_index, st_table *index_lookup_table, rb_ast_id_table_t *local_table_for_iseq, pm_scope_node_t *scope_node)
 {
-    ID local = pm_constant_id_lookup(scope_node, constant_id);
+    ID local = constant_id == 0 ? 0 : pm_constant_id_lookup(scope_node, constant_id);
```

Comment:
This is going to add an invalid ID into the local table. Is this necessary at all? It seems like we should probably skip this step entirely if there is no name on the block parameter.

Reply:
Good point! I changed it to skip writing to the local table at all if the `constant_id` is 0.